File: Semestru II/Tehnologii_Web/Laborator/Lab7/xml-dom/procesari/python/dom.py
#!/bin/python

# Program care realizeaza proceseaza un document XML prin DOM
# Autor: Sabin-Corneliu Buraga -- https://profs.info.uaic.ro/~busaco/
# Ultima actualizare: 20 martie 2009

import urllib
import xml.dom

from xml.dom.minidom import parse

print 'Incercam sa procesam documentul XML...'

try:
   # incarcam documentul
   doc = urllib.urlopen('projects.xml')
   # il procesam
   dom = parse(doc)
   # preluam elementul-radacina
   radacina = dom.documentElement
   print 'Elementul radacina este <' + radacina.tagName + '>'
   # parcurgem elementele <project> si oferim informatii despre fiecare
   for proiect in dom.getElementsByTagName('project'):
      print 'Proiectul ' + proiect.childNodes[1].firstChild.nodeValue + \
            ' este de clasa: ' + proiect.getAttribute('class')
except Exception:
   print 'Din pacate, a aparut o exceptie :('

File: Semestru II/Tehnologii_Web/Laborator/Lab7/xml-dom/procesari/perl/dom.pl
#!/usr/bin/perl

# Script Perl care contorizeaza proiectele de clasa A
# Autor: Sabin-Corneliu Buraga
# Ultima actualizare: 30 decembrie 2004
use XML::DOM;

$parser = new XML::DOM::Parser;
$doc = $parser->parsefile ('projects.xml');

$proiecte = $doc->getElementsByTagName ('project');
$nr_proiecte = $proiecte->getLength;

$proiecte_clasaA = 0; # contorizeaza proiectele 'A'

# baleiam toate proiectele
for (my $i = 0; $i < $nr_proiecte; $i++) {
  $proiect = $proiecte->item ($i);
  # preluam clasa proiectului
  $clasa = $proiect->getAttributeNode ('class')->getValue;
  $proiecte_clasaA++ if $clasa eq 'A';
}

# afisam la iesirea standard
print "Pentru documentul XML de mai jos:\n";
print $doc->toString;
print "\nSunt $proiecte_clasaA proiecte de clasa A.\n";

# eliberam memoria
$doc->dispose();
